juju/juju
# feat(cmd-register): prevent replacing existing controller if logged in

This PR adds additional checks to `juju register` to block users from adding a known controller.  The changes proposed here:
* prevent users from hitting [lp2073741](https://bugs.launchpad.net/juju/+bug/2073741?comments=all) by adding validation coverage
* improve the UX in some failure cases by raising the error before consuming the user's registration token and/or before we prompt them for a password

## Checklist




## QA steps

On computer 1 (with a juju controller):
```
juju add-user u1
juju add-user u2
````

On computer 2 (without a juju controller):
```
juju register TOKEN_FROM_u1
>> works
juju register TOKEN_FROM_u2 [--replace]
>> error with instructions.  This should not burn `TOKEN_FROM_u2`
```

## Documentation changes

Minor change in CLI help

## Links

**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/2073741
